shared/chat/conversation/input-area/normal/platform-input.desktop.js

```diff
@@ -215,143 +210,154 @@ class PlatformInput extends Component<PlatformInputProps & OverlayParentProps, S
-            />
-            {flags.explodingMessagesEnabled &&
-              this.props.isExploding &&
-              !this.props.isEditing &&
-              !this.state.hasText && (
-                <Icon
-                  color={globalColors.black_20}
-                  fontSize={34}
-                  hoverColor={globalColors.black_20}
-                  onClick={this._inputFocus}
-                  style={styleBoomIcon}
-                  type="iconfont-boom"
-                />
-              )}
-            {flags.explodingMessagesEnabled &&
-              this.props.showingMenu && (
-                <SetExplodingMessagePopup
-                  attachTo={this.props.attachmentRef}
-                  conversationIDKey={this.props.conversationIDKey}
-                  onAfterSelect={this._inputFocus}
-                  onHidden={this.props.toggleShowingMenu}
-                  visible={this.props.showingMenu}
+        {this.props.channelMentionPopupOpen && <MentionCatcher onClick={this._channelMentionCatcherClick} />}
+        {this.props.channelMentionPopupOpen && (
+          <ChannelMentionHud
+            conversationIDKey={this.props.conversationIDKey}
+            selectDownCounter={this.props.downArrowCounter}
+            selectUpCounter={this.props.upArrowCounter}
+            pickSelectedChannelCounter={this.props.pickSelectedCounter}
+            onPickChannel={this.props.insertChannelMention}
+            onSelectChannel={this.props.switchChannelMention}
+            filter={this.props.channelMentionFilter}
+          />
+        )}
+        <Kb.Box
+          style={collapseStyles([
+            styles.inputWrapper,
+            {
+              backgroundColor: this.props.isEditing ? globalColors.yellow3 : globalColors.white,
+              borderColor: this.props.explodingModeSeconds
+                ? globalColors.black_75
+                : this.props.isEditing
+                  ? globalColors.black_20
+                  : globalColors.black_20,
+            },
+          ])}
+        >
+          {!this.props.isEditing && (
+            <HoverBox
+              className={this.props.showingMenu ? 'expanded' : ''}
+              onClick={this._toggleShowingMenu}
+              ref={this.props.setAttachmentRef}
+              style={collapseStyles([
+                styles.explodingIconContainer,
+                {
+                  backgroundColor: this.props.explodingModeSeconds
+                    ? globalColors.black_75
+                    : globalColors.white,
+                },
+              ])}
+            >
+              {this.props.explodingModeSeconds ? (
+                <Kb.Text type="BodyTinyBold" style={styles.time}>
+                  {formatDurationShort(this.props.explodingModeSeconds * 1000)}
+                </Kb.Text>
+              ) : (
+                <Kb.Icon
+                  className="timer"
+                  onClick={this.props.isEditing ? undefined : this._toggleShowingMenu}
```

Comment:
is this ternary needed? this doesn't render if `!isEditing`

Reply:
Another testing remnant. Thanks!